### PR TITLE
Auth hook implementation

### PR DESCRIPTION
**Pull request by lemohammed**
I added some flexibility in how we route after the user has logged in.

Test it and let me know - it's lowkey almost 1 am so goodnight.

### CODE REVIEW

**Review by rjb75 — CHANGES_REQUESTED** [review]
Looks like ESLint should be ran locally

**Review by MrClean-1 — DISMISSED** [review]
Looks good mohammed, 1am coding is the best type of coding ;)

File: src/App.tsx [hunk 1]
```diff
@@ -41,46 +41,87 @@ import '@ionic/react/css/display.css';
...
-                <Route path="/register" component={RegisterLanding} exact={true} />
-                <Route path="/emailRegister" component={RegisterForm} exact={true} />
-                <Route path="/" render={() => <Redirect to="/landing" />} exact={true} />
-            </IonRouterOutlet>
-        </IonReactRouter>
-    </IonApp>
-);
+const App: React.FC = () => {
+    /**
+     * @author Mohamad Abdel Rida
```
Comment (RatikKapoor) [inline src/App.tsx:49]: 🌟 for documentation :)

File: src/pages/Login.tsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ import User from '../Models/User';
 async function handleSubmit(email: string, password: string) {
     if (await User.login(email, password)) {
         alert('login success');
+        window.history.replaceState({}, '', '/tab1'); //TODO Redirect use to the homepage
```
Comment (RatikKapoor) [inline src/pages/Login.tsx:27]: @MrClean-1 @lemohammed: Does this need to redirect to `/home` or `/landing`?

**Review by RatikKapoor — COMMENTED** [review]
Check my inline feedback on `Login.tsx`. If that's good then I'll go ahead and approve

File: src/pages/Login.tsx (hunk 2 above)
Comment (lemohammed) [reply src/pages/Login.tsx:27]: That needs to be to /home. Let me change it.
I had something similar in App.tsx

**Review by rjb75 — APPROVED** [review]

**Review by rjb75 — APPROVED** [review]
Looks good 🔐

**Review by RatikKapoor — APPROVED** [review]
💯 Thanks for the changes :)

**Pull request merged**